Refactor HOW-TO-MAKE-THESE.md: Add quick-sharun guide and practical examples (#232)
pkgforge-dev/Anylinux-AppImages: HOW-TO-MAKE-THESE.md

```

**Using debloated packages** (smaller AppImages):

```bash
EXTRA_PACKAGES="https://raw.githubusercontent.com/pkgforge-dev/Anylinux-AppImages/refs/heads/main/useful-tools/get-debloated-pkgs.sh"

wget "$EXTRA_PACKAGES" -O ./get-debloated-pkgs.sh
chmod +x ./get-debloated-pkgs.sh

# Installs a debloated MESA, vulkan, Qt, GTK, libicudata, and more
./get-debloated-pkgs.sh --add-mesa --prefer-nano

# Some appsm might requiere these as well
./get-debloated-pkgs.sh --add-common --prefer-nano ffmpeg-mini intel-media-driver-mini
```

-----------------------------------

| [Back to Index](#index) |
| - |

-----------------------------------

### *Using hooks*

Hooks are scripts that solve common problems automatically. Add them using the `ADD_HOOKS` variable:

```bash
export ADD_HOOKS="self-updater.bg.hook:fix-namespaces.hook"
./quick-sharun /usr/bin/myapp
```

**Available hooks:**

- **`self-updater.bg.hook`** - Makes the AppImage self-updatable using appimageupdatetool
- **`fix-namespaces.hook`** - Fixes namespace restrictions for apps that need them (web browsers and electron apps mostly)
- **`udev-installer.hook`** - Prompts to user to install bundled udev rules when needed

> - **`vulkan-check.src.hook`** - Checks and fixes several commons issues that might affect vulkan and hardware accel in general

Reviewer: ```suggestion
- **`vulkan-check.src.hook`** - Checks and fixes several commons issues that might affect vulkan and hardware acceleration in general
```

	"$CURRENTDIR"/bin/app "$@"
```

> Now we are ready to start making our truly portable AppImage, now just need to bundle the libraries and dynamic linker and we are good to go! Kinda now we need to fix the following issue… **And also bundling all the libraries needed isn't as easy as just running `ldd` + `cp`** Sharun handles this automatically (see below).

Reviewer: ```suggestion
We need to bundle the libraries and dynamic linker and we are **almost** good to go! However, to be fully ready, we need to fix the following issues below… **Bundling all the needed libraries isn't as easy as just running `ldd` + `cp`**, so we need some more robust solution. Sharun handles this automatically (see below).
```

### *Sharun*

There is a solution for this, made by @VHSGunzo called sharun:

https://github.com/VHSgunzo/sharun

> * sharun is able to find all the libraries your application needs, **including those that are dlopened**, it turns out a lot of applications depend on dlopened libraries, those are libraries you cannot easily find with just `ldd`. Sharun uses a deployment script called `lib4bin` that has the strace mode, **that mode makes `lib4bin` open the application with strace to check all the dlopened libraries and then bundle them.**

Reviewer: ```suggestion
* sharun is able to find all the libraries that your application needs, **including those that are dlopened**. It turns out that a lot of applications depend on dlopened libraries; those are the libraries that you cannot easily find with just `ldd`. Sharun uses a deployment script called `lib4bin` that has the strace mode; **that mode makes `lib4bin` open the application with strace to check all the dlopened libraries and then bundle them.**
```

> * sharun is also just not for AppImages, you can use it anywhere you need to make any sort of application portable, you can even make pseudo static binaries from existing dynamic binaries which sharun does with the help of wrappe.

Reviewer: ok, then ommit that `the`
